### PR TITLE
Geom: Triangle

### LINKED ISSUE #958: Geom: Implement Shape hierarchy
### Description of the task
Part of the geom redesign is implementing the following Shape hierarchy:

- Shape (i)
    - ConvexShape (i)
      - Circle
      - ConvexPolygon
   - Polygon
      - ConvexPolygon
         - Triangle
         - Rectangle
Notes: 
- ConvexPolygon inherits from both ConvexShape and Polygon
- Methods that use these belong in util and will be in a separate issue

### Acceptance criteria

- [ ] Implement the hierarchy as above
- [ ] Fields getters and setters for each class/interface

### Blocked By
- #956 Shapes depend on points / vectors(?)
- #957 Shapes depend on splines rays and polynomials

### PR DESCRIPTION
**Pull request by scveloso**
### Description
Small PR to add triangle to shape hierarchy in ```new_geom```

### Testing Done
Added unit tests for triangle

### Resolved Issues
resolves #958

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by scveloso** [comment]
A couple of questions/notes:
- Should we prevent construction of degenerate Triangles? Old geom did not do this
- ```center()``` was the only method I saw in-use specifically to Triangles and it is implemented in ```indirect_chip.cpp```, would make it cleaner to move it to ```new_geom```
- Let me know if there are other methods that ```Triangle``` should implement

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
New hierarchy is so clean :heart_eyes:. Only comment is maybe we should start breaking shapes out into their own libraries so we don't have a monolithic `new_geom` lib? I think this was brought up in one of @jsn0716 's previous PRs

**Comment by garethellis0** [comment]
> Only comment is maybe we should start breaking shapes out into their own libraries so we don't have a monolithic new_geom lib?

:+1: for this

**Comment by MathewMacDougall** [comment]
> Should we prevent construction of degenerate Triangles? Old geom did not do this

I'm fine not doing this, and I think if we do we'd likely want to do it for all Polygons

> center() was the only method I saw in-use specifically to Triangles and it is implemented in indirect_chip.cpp, would make it cleaner to move it to new_geom

:+1:

> Let me know if there are other methods that Triangle should implement

`center` and `area` are all I think are needed for now

**Comment by jsn0716** [comment]
Formatting check seems to be failing, not much else to add to the other comments

**Comment by scveloso** [comment]
How were you guys thinking of breaking up shapes in their own libraries? 

Do you mean one directory for each shape? ```new_geom/triangle/triangle.h``` ```new_geom/triangle/triangle.cpp``` ```new_geom/triangle/triangle_test```? 

I can do it in this PR since its relatively small

File: src/scveloso [hunk 1]
```diff
@@ -0,0 +1 @@
+scveloso
```
Comment (jsn0716) [inline src/scveloso]: Don't think this is meant to be here

**Review by jsn0716 — CHANGES_REQUESTED** [review]
> How were you guys thinking of breaking up shapes in their own libraries?
> 
> Do you mean one directory for each shape? `new_geom/triangle/triangle.h` `new_geom/triangle/triangle.cpp` `new_geom/triangle/triangle_test`?
> 
> I can do it in this PR since its relatively small

I believe this was meant for the `BUILD` file, to break up the `cc_library` similar to how it was done in #1094

**Comment by MathewMacDougall** [comment]
Yup just different `cc_library` targets for each shape / geometry type, no need for more folders

**Review by jonathanlew — DISMISSED** [review]
LGTM!

**Review by MathewMacDougall — DISMISSED** [review]
:fire: LGTM

**Comment by jsn0716** [comment]
LGTM but CI build is failing

**Review by garethellis0 — COMMENTED** [review]
LGTM pending CI

**Comment by scveloso** [comment]
Something went wrong with my linux machine, sorry for the delay on this one

**Review by jonathanlew — APPROVED** [review]

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by MathewMacDougall — APPROVED** [review]

**Review by jsn0716 — APPROVED** [review]

**Pull request merged**